payment/payment_test.go

```diff
@@ -29,3 +29,12 @@ func TestSegwitAddress(t *testing.T) {
 		t.Errorf("TestSegwitAddress: error when encoding segwit")
 	}
 }
+
+func TestScriptHash(t *testing.T) {
+	_, publicKey := btcec.PrivKeyFromBytes(btcec.S256(), privateKeyBytes)
+	p2wpkh := payment.FromPublicKey(publicKey, &network.Regtest)
+	pay := payment.FromPayment(&p2wpkh)
```

Comment:
When running the tests, if you try to log the `Script` that is hashed within the `FromPayment` you can see that it is an empty byte slice.
Since the `hash160` function accepts even empty byte slices it seems that this is working as expected, but it is not actually.

We should add some changes to the `FromPublicKey` method to make this work correctly.

I just opened a ticket for this, hope you can find it clear.

@sekulicd

Reply:
Please @altafan can you review this after changes?